src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -807,17 +820,18 @@ private Compilation CreateEmptyCompilation()
                     }
 
                     // We will finalize the compilation by adding full contents here.
-                    var (compilationWithGeneratedDocuments, nextGeneratorInfo) = await AddExistingOrComputeNewGeneratorInfoAsync(
+                    var (compilationWithGeneratedDocuments, generatedDocuments, generatorDriver) = await AddExistingOrComputeNewGeneratorInfoAsync(
```

Comment:
i found thsi much clearer to just return the constituent data, not the full generatorinfo.  this way we could have *this* be the location where we say `WithDocsAreFinal(true)` instead of having to ensure that *all* paths called from AddExistingOrComputeNewGeneratorInfoAsync set this bit.